pandas/_testing/asserters.py

```diff
@@ -312,6 +315,15 @@ def _check_types(left, right, obj="Index"):
         if not exact:
             return
 
+        ldtype = left.dtype
+        rdtype = right.dtype
+
+        if exact == "equiv" and is_numeric_dtype(ldtype) and is_numeric_dtype(rdtype):
```

Comment:
hmm, can you move this to assert_class_equal (or actually where equiv is now checked) rather than here

Reply:
This was changed in #41980, and there's no changes in asserters.py in this PR anymore, so I'm closing this comment.